pkg/controller/mutate_aks.go

```diff
@@ -13,64 +13,85 @@ import (
 
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/azure"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/events"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
 )
 
+type ipResource string
+type ipAddress string
+
 // MutateAKS applies changes to Kubernetes resources.
-func (c AppGwIngressController) MutateAKS(event events.Event) error {
+func (c AppGwIngressController) MutateAKS() error {
+	appGw, cbCtx, err := c.getAppGw()
+	if err != nil {
+		return err
+	}
+
+	// update all ingresses with IP address obtained from existing App Gateway configuration
+	for _, ingress := range cbCtx.IngressList {
+		c.updateIngressStatus(appGw, cbCtx, ingress)
+	}
 	return nil
 }
 
-func (c AppGwIngressController) updateIngressStatus(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, event events.Event) {
-	ingress, ok := event.Value.(*v1beta1.Ingress)
-	if !ok {
-		return
-	}
+func (c AppGwIngressController) updateIngressStatus(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, ingress *v1beta1.Ingress) {
+	ips := getIPs(appGw, c.azClient)
```

Comment:
this can be moved 1 level up to `MutateAKS()`

Reply:
Good idea!